### PR TITLE
Added `total_effective_plot_size` for `get_harvesters` RPC API

### PR DESCRIPTION
**Pull request by ChiaMineJP**
This PR is a partial PR cut from cmj.plot2_0 branch.

### CODE REVIEW

File: chia/cmds/farm_funcs.py [hunk 1]
```diff
@@ -132,10 +133,15 @@ def process_harvesters(harvester_peers_in: Dict[str, Dict[str, Any]]) -> None:
...
                 else:
                     total_plot_size_harvester = harvester_dict["total_plot_size"]
+                    total_effective_plot_size_harvester = harvester_dict["total_effective_plot_size"]
                     plot_count_harvester = harvester_dict["plots"]
                     PlotStats.total_plot_size += total_plot_size_harvester
+                    PlotStats.total_effective_plot_size += total_effective_plot_size_harvester
                     PlotStats.total_plots += plot_count_harvester
-                    print(f"   {plot_count_harvester} plots of size: {format_bytes(total_plot_size_harvester)}")
+                    print(
+                        f"   {plot_count_harvester} plots of size: {format_bytes(total_plot_size_harvester)} raw, "
```
Comment (arvidn) [inline chia/cmds/farm_funcs.py:142]: I think "on-disk" is better than "raw". that's what it means, right?

File: chia/plot_sync/receiver.py [hunk 2]
```diff
@@ -82,6 +83,7 @@ class Receiver:
     _keys_missing: List[str]
     _duplicates: List[str]
     _total_plot_size: int
+    _total_effective_plot_size: float
```
Comment (arvidn) [inline chia/plot_sync/receiver.py:86]: it seems odd to make this a `float`. is there a good reason for that?
when adding these up, you may have non-deterministic results

File: chia/plot_sync/receiver.py [hunk 3]
```diff
@@ -329,6 +336,9 @@ async def _sync_done(self, data: PlotSyncDone) -> None:
         self._keys_missing = self._current_sync.delta.keys_missing.additions.copy()
         self._duplicates = self._current_sync.delta.duplicates.additions.copy()
         self._total_plot_size = sum(plot.file_size for plot in self._plots.values())
+        self._total_effective_plot_size = sum(
+            UI_ACTUAL_SPACE_CONSTANT_FACTOR * int(_expected_plot_size(plot.size)) for plot in self._plots.values()
```
Comment (arvidn) [inline chia/plot_sync/receiver.py:340]: how come `_expected_plot_size()` doesn't return the effective size? is "expected" different from "effective"?
I would expect the effective size depends on the compression level, does it not?

**Review by arvidn — DISMISSED** [review]

**Comment by wallentx** [comment]
plot_sync test failures on this appear to be real

File: chia/plot_sync/receiver.py (hunk 2 above)
Comment (ChiaMineJP) [reply chia/plot_sync/receiver.py]: Right, I changed this to `int`.

File: chia/plot_sync/receiver.py (hunk 3 above)
Comment (ChiaMineJP) [reply chia/plot_sync/receiver.py:340]: `_expected_plot_size()` is a part of concensus, meaning it is used to calculate `require_iter` and it shouldn't depend on compression level. The difference between actual plot file size and effective plot size (roughly 101GiB) is what user earns by compression.

File: chia/cmds/farm_funcs.py (hunk 1 above)
Comment (ChiaMineJP) [reply chia/cmds/farm_funcs.py:142]: Thanks! I'll change the text.

**Comment by emlowe** [comment]
Still seem problems in `plot_sync` tests:



```
21:36:50 chia.plot_sync.sender: WARNING set_response skip unexpected response: PlotSyncResponse: identifier {'message_id': 0, 'sync_id': 1689629810, 'timestamp': 1689629810}, message_type 78, error None
21:36:50 chia.plot_sync.sender: WARNING set_response skip unexpected response: PlotSyncResponse: identifier {'message_id': 1, 'sync_id': 1689629810, 'timestamp': 1689629810}, message_type 79, error None
21:36:50 chia.plot_sync.sender: WARNING set_response skip unexpected response: PlotSyncResponse: identifier {'message_id': 2, 'sync_id': 1689629810, 'timestamp': 1689629810}, message_type 79, error None
21:36:50 chia.plot_sync.sender: WARNING set_response skip unexpected response: PlotSyncResponse: identifier {'message_id': 3, 'sync_id': 1689629810, 'timestamp': 1689629810}, message_type 80, error None
21:36:51 chia.plot_sync.sender: WARNING set_response skip unexpected response: PlotSyncResponse: identifier {'message_id': 4, 'sync_id': 1689629810, 'timestamp': 1689629811}, message_type 81, error None
21:36:51 chia.plot_sync.sender: WARNING set_response skip unexpected response: PlotSyncResponse: identifier {'message_id': 5, 'sync_id': 1689629810, 'timestamp': 1689629811}, message_type 82, error None
21:36:51 chia.plot_sync.sender: WARNING set_response skip unexpected response: PlotSyncResponse: identifier {'message_id': 6, 'sync_id': 1689629810, 'timestamp': 1689629811}, message_type 83, error None
21:36:51 chia.plot_sync.receiver: WARNING _process: node_id faa2ae556f37493063eabe85219f1eabbe26684032092fc5c852535ffe4e91b1, Exception Value 3[429](https://github.com/Chia-Network/chia-blockchain/actions/runs/5579891107/jobs/10198151500?pr=15770#step:16:430)9414762053697536 does not fit into uint64
21:36:51 chia.plot_sync.sender: WARNING set_response skip unexpected response: PlotSyncResponse: identifier {'message_id': 7, 'sync_id': 1689629810, 'timestamp': 1689629811}, message_type 84, error PlotSyncError: code -1, count Value 34299414762053697536 does not fit into uint64, expected_identifier None
```

**Comment by ChiaMineJP** [comment]
@emlowe I've found out the root issue.

image

The code above can set k_size=254 to `DiskProver`.
When `k_size = 254`, it overflows `uint64` when calculating `_expected_plot_size(k_size)`

```python
def _expected_plot_size(k: int) -> uint64:
    return uint64(((2 * k) + 1) * (2 ** (k - 1)))
```

`(2 ** (k - 1))` in the above calculation clearly overflows `uint64` when `k=254`.

I fixed the test code and pushed a commit.

**Review by emlowe — APPROVED** [review]

**Pull request merged**